src/server/appserver.cpp

```diff
@@ -2177,36 +2206,51 @@ void AppServer::onShowWindowsUpdateErrorDialog() {
 }
 
 void AppServer::onRestartClientReceived() {
-    // Check last start time
-    if (clientCrashedRecently()) {
-        LOG_FATAL(_logger, "Client crashed twice in a short time, exiting");
-        bool found = false;
-        if (!KDC::ParmsDb::instance()->updateAppState(AppStateKey::LastClientSelfRestartDate, std::string("0"), found) ||
-            !found) {
-            addError(Error(errId(), ExitCode::DbError, ExitCause::DbEntryNotFound));
-            LOG_WARN(_logger, "Error in ParmsDb::updateAppState");
+#if NDEBUG
+    if (clientHasCrashed()) {
+        LOG_ERROR(_logger, "Client disconnected (has crashed)");
+        if (clientCrashedRecently()) {
+            LOG_FATAL(_logger, "Client has crashed twice in a short time, exiting");
```

Comment:
**Suggestion:** write a small function named `handleRecentCrash` that encapsulates the logic followed in this special case. This could be unit tested by means of a mock of `ParmsDb`or by checking `ParmsDb` content directly.

Reply:
Done